### PR TITLE
test: Port tag_keys query_tests to end-to-end tests

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to https://github.com/influxdata/influxdb_iox/issues/6617.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        request: GrpcRequestBuilder::new()
+            .timestamp_range(150, 201)
+            .tag_predicate("state", "MA"),
+        expected_keys: vec!["_m(0x00)", "city", "state", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:182]: Was `list_tag_columns_measurement_name_and_predicate` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/tag_keys.rs#L145-L150

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        request: GrpcRequestBuilder::new()
+            .measurement_predicate("o2")
+            .tag_predicate("state", "NY"),
+        expected_keys: vec!["_m(0x00)", "borough", "city", "state", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:195]: Was `list_tag_columns_measurement_name_and_predicate_and_timestamp` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/tag_keys.rs#L153-L160

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            .timestamp_range(1, 550)
+            .tag_predicate("state", "NY")
+            .measurement_predicate("o2"),
+        expected_keys: vec!["_m(0x00)", "city", "state", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:209]: Was `list_tag_name_end_to_end` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/tag_keys.rs#L163-L170

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    Arc::new(TagKeysTest {
+        setup_name: "TwoMeasurementsManyNulls",
+        request: GrpcRequestBuilder::new().timestamp_range(150, 201),
+        expected_keys: vec!["_m(0x00)", "city", "state", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:86]: I redid the tests involving max_time that were https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/tag_keys.rs#L173-L207

and instead made them match [the measurement_names ones](https://github.com/influxdata/influxdb_iox/pull/6757); these were a little different and didn't look quite right to me.

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+        request: GrpcRequestBuilder::new()
+            .timestamp_range(0, 10000)
+            .tag_predicate("host", "server01"),
+        expected_keys: vec!["_m(0x00)", "host", "name", "region", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:222]: Was `list_tag_name_with_periods` https://github.com/influxdata/influxdb_iox/blob/5b14caa780b77c6607c84ad38578ed4d28f15e86/query_tests/src/influxrpc/tag_keys.rs#L210-L216

**Review by carols10cents — COMMENTED** [review]
Matching up

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    Arc::new(TagKeysTest {
+        setup_name: "MeasurementWithMaxTime",
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME, MAX_NANO_TIME + 1),
+        expected_keys: vec!["_m(0x00)", "host", "_f(0xff)"],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
[the remark this answers is not in the document]
Comment (alamb) [reply influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/tag_keys.rs:120]: this threw me as it was out of order 👍

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you very much @carols10cents

**Pull request merged**